rust/src/sip/parser.rs

```diff
@@ -97,6 +97,22 @@ fn is_header_value(b: u8) -> bool {
     is_alphanumeric(b) || is_token_char(b) || b"\"#$&(),/;:<=>?@[]{}()^|~\\\t\n\r ".contains(&b)
 }
 
+fn expand_header_name(h: &str) -> &str {
+    match h {
+        "i" => "Call-ID",
```

Comment:
Can we detect if some header uses the short or long form ?

Reply:
Why should we detect that?

Reply:
To get the most expressivity.
I do no say we should do that, I just asked if we can.

I guess this deserves some mention in the docs., like
```
This keyword matches on the From field that can be present in SIP headers.
It matches both short and regular form, and these forms cannot be distinguished by this keyword.
```

Reply:
I'm not sure I understand what you mean. If a header uses the short form, it is 'expanded' ( converted to the long form) to enable matching in both cases with its related sticky buffer.
If this is what you are referring to, it has already been implemented.
 If not, I'm unclear on what you mean by 'detecting which form is used by a header.'

Reply:
I meant 2 things :

> If a header uses the short form, it is 'expanded' ( converted to the long form) to enable matching in both cases with its related sticky buffer.

1. This should be in the docs

2. And the doc should also mention that there is only the normalized form, not the raw one (as we can have for http uri for instance where there are 2 keywords)

Reply:
> I meant 2 things :
> 
> > If a header uses the short form, it is 'expanded' ( converted to the long form) to enable matching in both cases with its related sticky buffer.
> 
>     1. This should be in the docs
> 
>     2. And the doc should also mention that there is only the normalized form, not the raw one (as we can have for http uri for instance where there are 2 keywords)

+1